### ISSUE TITLE
MobilityDB installation

### ISSUE THREAD

**Issue by TaehoonK** [aistairc/pygeoapi-mf-api#5]
**Is your feature request related to a problem? Please describe.**

I'd like to know about the installation of Pygeoapi for OGC API-Moving Features.
Are there any modifications to it?
For example, did you consider adding some commands to install MobilityDB for Docker?
You already used some packages to use MobilityDB in the Python file, but did you add these requirements to the Python requirements file, such as requirements.txt or requirements-provider.txt?
When I checked your branch (‘mf-api-updates’), there were no modifications to this issue.
However, this issue is essential functionality, and it should be resolved.

**Describe the solution you'd like**

1. Add some commands in the Docker file for installing MobilityDB.
2. Add used Python packages to the requirements.txt or requirements-privider.txt. It depends on the used code location. 
3. Test it with a clean installation. 

**Describe alternatives you've considered**

**Additional context**

**Comment by ogi-ts-shimizu** [comment]
@TaehoonK 
Regarding the Python package, I apologize for forgetting to commit it. I have now added 'pymeos' to requirements-provider.txt.

As for adding commands to the Dockerfile, I'm not experienced with Docker environment setup, so I'm unsure what commands to add.  I checked the mf-api Dockerfile (https://github.com/aistairc/mf-api/blob/main/Dockerfile) and saw that the base image is mobilitydb/mobilitydb:14-3.2-1. I then looked at the Dockerfile for that base image (https://github.com/MobilityDB/MobilityDB-docker/blob/master/14-3.2-1/Dockerfile).

My understanding is that I should add the following commands to the Dockerfile. Is this correct?
` # Install MobilityDB `
`RUN wget -O MobilityDB.tar.gz "https://github.com/MobilityDB/MobilityDB/archive/v${MOBILITYDB_VERSION}.tar.gz" \`
 `&& mkdir -p /usr/local/src/MobilityDB \`
 `&& tar \`
        `--extract \`
        `--file MobilityDB.tar.gz \`
       ` --directory /usr/local/src/MobilityDB \`
        `--strip-components 1 \`
 `&& rm MobilityDB.tar.gz`
`RUN mkdir /usr/local/src/MobilityDB/build`
`RUN cd /usr/local/src/MobilityDB/build && \`
    `cmake .. && \`
    `make -j$(nproc) && \`
    `make install`
`RUN rm /docker-entrypoint-initdb.d/mobilitydb.sh`
`COPY sql/initdb-mobilitydb.sh /docker-entrypoint-initdb.d/mobilitydb.sh`
`RUN chmod +x /docker-entrypoint-initdb.d/mobilitydb.sh`

**Comment by ogi-ts-shimizu** [comment]
The Docker environment has been built and verified to work.
The following command confirms Docker startup and pytest completion.

```
git clone -b mf-api-updates https://github.com/ogi-ts-shimizu/pygeoapi-ogi-mf-api.git
docker compose build
docker-compose up
docker exec pygeoapi-ogi-mf-api-mf-api-1 pytest ./tests/test_postgresql_mobilitydb.py
```

In Windows environment, the newline code may be converted to CRLF in Git clone.
In this case, please change the newline code to LF for the following files in the docker folder.
・entrypoint.sh
・initdb-mobilitydb.sh

For the first parameter of the docker exec command, specify the name of the Docker Desktop container.
![image](https://github.com/user-attachments/assets/f6a2ce1b-24e2-463d-960b-e7aa3290ac78)

**Comment by TaehoonK** [comment]
Thank you for your contribution. 
I will test it as soon as possible. 
But I think DockerfileMobiitydb will be moved to [pygeoapi-examples](https://github.com/geopython/pygeoapi-examples).

**Comment by breeze0512** [comment]
Hello @ogi-ts-shimizu , this is Wijae Cho, working with Taehoon. 
An issue related to installing pymeos occurred during the docker compose build process. 
(To be precise, the issue occurs while installing the dependent library, pymeos-cffi.)
I will attach an image with the details of the issue below. 
![mf-api-build-issue](https://github.com/user-attachments/assets/a5d9be09-c170-4fde-b8f9-2d2a59aa234f)

If you know the solution, we would appreciate it if you could share it. 
Thank you.

**Comment by TaehoonK** [comment]
@ogi-ts-shimizu 
If you have successfully created a docker image and container, please share it as I did before (note 1) by referring to the link below (note 2). 
note 1: https://github.com/TaehoonK/mf-api/pkgs/container/mf-api-server 
note 2: https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry

**Comment by ogi-ts-shimizu** [comment]
We will try to push the docker image with the URL you shared.
In parallel, we will check if the same error occurs on another PC.

**Comment by ogi-ts-shimizu** [comment]
@breeze0512 
Could you check the docker architecture with the following command?
`docker info`
In my environment, it is x86_64.
![image](https://github.com/user-attachments/assets/bb6ff9b9-54dd-44ff-b682-5209b6e4dabe)

It seems that CPU architecture differences may cause build errors.
https://qiita.com/inetcpl/items/30e2d888eccc1df62023

Can you try again with the following options to the build command?
--platform linux/amd64

`docker build --platform linux/amd64 .`

**Comment by breeze0512** [comment]
@ogi-ts-shimizu 
The method you provided works without triggering the issue I mentioned yesterday. 
However, since the `5000` port used by mf-api is already in use on my PC, I’ll need to use a different port, such as 5050. (It seems that other users may also face issues if the 5000 port is already in use on their systems.)
Do you know of an easy way to manage this?

**Comment by ogi-ts-shimizu** [comment]
@breeze0512 
It seems that the only way to do this is to have the user change the port in docker-compose.yml to match the user's environment.
![image](https://github.com/user-attachments/assets/ea751bbf-5742-4855-a8b0-43f8c20f3d0f)
I think it would be good to have the port as 5050 for once.
Please change the port to 5050 and build again.

**Comment by ogi-ts-shimizu** [comment]
@TaehoonK  @breeze0512 

We have pushed the Docker image to Github using the URL you shared.

https://github.com/ogi-ts-shimizu/pygeoapi-ogi-mf-api/pkgs/container/pygeoapi-mf-api
https://github.com/ogi-ts-shimizu/pygeoapi-ogi-mf-api/pkgs/container/pygeoapi-mf-api-mobilitydb

The following commands are used to build and start the Docker environment and run pytest.
```
docker pull ghcr.io/ogi-ts-shimizu/pygeoapi-mf-api:1.0

docker pull ghcr.io/ogi-ts-shimizu/pygeoapi-mf-api-mobilitydb:1.0

docker network create pygeoapi-mf-api-network

docker run -p 5050:80 -d --net pygeoapi-mf-api-network --name pygeoapi-mf-api ghcr.io/ogi-ts-shimizu/pygeoapi-mf-api:1.0

docker run -d --net pygeoapi-mf-api-network -p 25432:5432 --name mobilitydb -e POSTGRES_DB=mobilitydb -e POSTGRES_USER=docker -e POSTGRES_PASSWORD=docker ghcr.io/ogi-ts-shimizu/pygeoapi-mf-api-mobilitydb:1.0

docker exec pygeoapi-mf-api pytest ./tests/test_postgresql_mobilitydb.py
```

Best regards

**Comment by TaehoonK** [comment]
@ogi-ts-shimizu 
We have some issues with running DockerfileMobilityDB: The DB server cannot start. 
We can resolve this issue by updating the base image 14-3.2 to 17-3.5.
Finally, there is no critical issue; we are investigating each functionality. 
Thank you for your quick support!

**Comment by ogi-ts-shimizu** [comment]
@TaehoonK 
Thanks for the confirmation and pointing out the DB issue.
Updated and pushed DockerfileMobilitydb from base image 14-3.2 to 17-3.5.
Also updated base image 14-3.2 to 17-3.5 and built Docker image(1.1)

We confirmed the operation with the following command with a different version.
```
docker pull ghcr.io/ogi-ts-shimizu/pygeoapi-mf-api:1.1

docker pull ghcr.io/ogi-ts-shimizu/pygeoapi-mf-api-mobilitydb:1.1

docker network create pygeoapi-mf-api-network

docker run -p 5050:80 -d --net pygeoapi-mf-api-network --name pygeoapi-mf-api ghcr.io/ogi-ts-shimizu/pygeoapi-mf-api:1.1

docker run -d --net pygeoapi-mf-api-network -p 25432:5432 --name mobilitydb -e POSTGRES_DB=mobilitydb -e POSTGRES_USER=docker -e POSTGRES_PASSWORD=docker ghcr.io/ogi-ts-shimizu/pygeoapi-mf-api-mobilitydb:1.1

docker exec pygeoapi-mf-api pytest ./tests/test_postgresql_mobilitydb.py

```
Best regards!

**Comment by TaehoonK** [comment]
This issue is resolved and merged into the main branch by a pull request (https://github.com/aistairc/pygeoapi-mf-api/pull/1)
